### PR TITLE
ipn/ipnlocal,ipnserver: rename {,Set}CurrentUser to {,Set}CurrentUserID

### PR DESCRIPTION
**Pull request by maisem**
Address comments from https://github.com/tailscale/tailscale/pull/6506#discussion_r1032454064

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnserver/server.go [hunk 1]
```diff
@@ -304,8 +304,14 @@ func (s *Server) checkConnIdentityLocked(ci *ipnauth.ConnIdentity) error {
 			return inUseOtherUserError{fmt.Errorf("Tailscale already in use by %s, pid %d", active.User().Username, active.Pid())}
 		}
 	}
-	if cu := s.b.CurrentUser(); cu != "" && cu != ci.UserID() {
-		return inUseOtherUserError{fmt.Errorf("Tailscale already in use by %s", s.b.CurrentUser())}
+	if cu := s.b.CurrentUserID(); cu != "" && cu != ci.UserID() {
+		// TODO(maisem): cache this?
+		u, err := user.LookupId(cu)
```
Comment (andrew-d) [inline ipn/ipnserver/server.go:309]: Should we use the `LookupUserByID` in ipnauth here?

Comment (bradfitz) [reply ipn/ipnserver/server.go:309]: yeah probably

Comment (maisem) [reply ipn/ipnserver/server.go]: Done

**Review by andrew-d — APPROVED** [review]

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -2053,7 +2053,12 @@ func (b *LocalBackend) CheckIPNConnectionAllowed(ci *ipnauth.ConnIdentity) error
 		return errors.New("empty user uid in connection identity")
 	}
 	if uid != serverModeUid {
-		return fmt.Errorf("Tailscale running in server mode (uid=%v); connection from %q not allowed", serverModeUid, uid)
+		u, err := ipnauth.LookupUserFromID(b.logf, serverModeUid)
+		if err != nil {
+			return fmt.Errorf("Tailscale running in server mode (uid=%q); connection from %q not allowed", serverModeUid, uid)
+		} else {
+			return fmt.Errorf("Tailscale running in server mode (user=%q); connection from %q not allowed", u.Username, uid)
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:2060]: these are still kinda gross with uids

Maybe make a little closure here that return a best effort stringification of a uid using ipnauth.LookupUserFromID and then use it in one fmt.Errorf return path for both the serverModeUid and uid?

**Review by bradfitz — APPROVED** [review]
LGTM otherwise

File: ipn/ipnlocal/local.go (hunk 2 above)
Comment (maisem) [reply ipn/ipnlocal/local.go]: added a method to LocalBackend instead of a closure.

**Pull request merged**